src/map/atcommand.c

```diff
@@ -7622,33 +7622,47 @@ ACMD(monsterignore)
  *------------------------------------------*/
 ACMD(fakename)
 {
-	if (!*message)
-	{
-		if (sd->fakename[0])
-		{
+	int option = 0; // Hide both guild/party name
+	char fakename[NAME_LENGTH];
+
+	if (!*message) {
+		if (sd->fakename[0]) {
 			sd->fakename[0] = '\0';
+			sd->fakename_option = 0;
 			clif->charnameack(0, &sd->bl);
-			if( sd->disguise )
+			if (sd->disguise)
 				clif->charnameack(sd->fd, &sd->bl);
-			clif->message(sd->fd, msg_fd(fd,1307)); // Returned to real name.
+			clif->message(sd->fd, msg_fd(fd, 1307)); // Returned to real name.
 			return true;
 		}
 
-		clif->message(sd->fd, msg_fd(fd,1308)); // You must enter a name.
+		clif->message(sd->fd, msg_fd(fd, 1308)); // Please enter an option number and name (usage: @fakename <0-3> <name>)
 		return false;
 	}
 
-	if (strlen(message) < 2)
-	{
-		clif->message(sd->fd, msg_fd(fd,1309)); // Fake name must be at least two characters.
+	memset(fakename, '\0', sizeof(fakename));
+
+	if (sscanf(message, "%d %23[^\n]", &option, fakename) < 2) {
+		clif->message(sd->fd, msg_fd(fd, 1308)); // Please enter an option number and name (usage: @fakename <0-3> <name>)
+		return false;
+	}
+
+	if (option < 0 || option > 3) {
+		clif->message(sd->fd, msg_fd(fd, 1308)); // Please enter an option number and name (usage: @fakename <0-3> <name>)
+		return false;
+	}
+
+	if (strlen(fakename) < 2) {
+		clif->message(sd->fd, msg_fd(fd, 1309)); // Fake name must be at least two characters.
```

Comment:
Any idea why this restriction was put(before)? Why can't fakename be of 1 character?